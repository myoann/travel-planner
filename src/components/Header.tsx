import Image from "next/image";

export default function Header() {
  return (
    <header className="sticky top-0 w-full h-16 z-50 flex justify-between items-center bg-[#fbfbfd] shadow-sm px-8 py-4">
      <a href="/" className="font-bold">
        PLAN MY TRIP
      </a>

      <Image src="/logo.png" alt="Yoann Moise Logo" width={40} height={40} />
    </header>
  );
}
