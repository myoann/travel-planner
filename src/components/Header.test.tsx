import { render, screen } from "@testing-library/react";

import Header from "./Header";

describe("Header", () => {
  it("matches the snapshot", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });

  it("has a a link which redirects to /", () => {
    render(<Header />);

    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "/");
  });

  it("has an image with the alt 'Yoann Moise Logo'", () => {
    render(<Header />);

    const image = screen.getByRole("img");
    expect(image).toHaveAttribute("alt", "Yoann Moise Logo");
  });
});
