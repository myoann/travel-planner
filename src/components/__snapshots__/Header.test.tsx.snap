// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<div>
  <header
    class="sticky top-0 w-full h-16 z-50 flex justify-between items-center bg-[#fbfbfd] shadow-sm px-8 py-4"
  >
    <a
      class="font-bold"
      href="/"
    >
      PLAN MY TRIP
    </a>
    <img
      alt="Yoann Moise Logo"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Flogo.png&w=96&q=75"
      srcset="/_next/image?url=%2Flogo.png&w=48&q=75 1x, /_next/image?url=%2Flogo.png&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </header>
</div>
`;
